### PR TITLE
Color migration guide

### PR DESCRIPTION
**Pull request by BD103**
Originally part of #1376, but I split it out due to the amount of content.

This is the draft migration guide for the new color module. Please review it, re-organize it, and correct it as much as you like!

### CODE REVIEW

**Comment by BD103** [comment]
Note to self: this should include migrations for https://github.com/bevyengine/bevy/pull/12163, https://github.com/bevyengine/bevy/pull/12079, https://github.com/bevyengine/bevy/pull/12105, https://github.com/bevyengine/bevy/pull/12116, https://github.com/bevyengine/bevy/pull/12114, https://github.com/bevyengine/bevy/pull/12115, https://github.com/bevyengine/bevy/pull/12168, https://github.com/bevyengine/bevy/pull/12583, and https://github.com/bevyengine/bevy/pull/12399.

**Review by viridia — APPROVED** [review]

File: release-content/0.14/migration-guides/12163_Migrate_from_LegacyColor_to_bevy_colorColor.md [hunk 1]
```diff
@@ -1,22 +1,142 @@
...
-
-These are quite similar! Both are enums storing a color in a specific color space (or to be more precise, using a specific color model). However, each of the different color models now has its own type.
-
-TODO…
-
-- `Color::rgba`, `Color::rgb`, `Color::rbga_u8`, `Color::rgb_u8`, `Color::rgb_from_array` are now `Color::srgba`, `Color::srgb`, `Color::srgba_u8`, `Color::srgb_u8` and `Color::srgb_from_array`.
-- `Color::set_a` and `Color::a` is now `Color::set_alpha` and `Color::alpha`. These are part of the `Alpha` trait in `bevy_color`.
-- `Color::is_fully_transparent` is now part of the `Alpha` trait in `bevy_color`
-- `Color::r`, `Color::set_r`, `Color::with_r` and the equivalents for `g`, `b` `h`, `s` and `l` have been removed due to causing silent relatively expensive conversions. Convert your `Color` into the desired color space, perform your operations there, and then convert it back into a polymorphic `Color` enum.
-- `Color::hex` is now `Srgba::hex`. Call `.into` or construct a `Color::Srgba` variant manually to convert it.
```
Comment (gagnus) [inline release-content/0.14/migration-guides/12163_Migrate_from_LegacyColor_to_bevy_colorColor.md:11]: or .to_srgba()

File: release-content/0.14/migration-guides/12163_Migrate_from_LegacyColor_to_bevy_colorColor.md [hunk 2]
```diff
@@ -1,22 +1,142 @@
...
+#### Alpha
+
+Alpha, also known as transparency, used to be referred to by the letter `a`. It is now called by its full name within structs and methods.
+
+- `Color::set_a` and `Color::a` are now `Color::set_alpha` and `Color::alpha`. These are part of the new `Alpha` trait.
+- Additionally, `Color::is_fully_transparent` is now part of the `Alpha`.
+
+#### CSS Constants
+
+The various CSS color constants are no longer stored directly on `Color`. Instead, they’re defined in the `Srgba` color space, and accessed via `bevy::color::palettes::css`. Call `.into()` on them to convert them into a `Color` for quick debugging use, and consider using the much prettier `tailwind` palette for prototyping. Furthermore, the `LIME_GREEN` color has been renamed to `LIMEGREEN` to comply with the standard naming.
```
Comment (gagnus) [inline release-content/0.14/migration-guides/12163_Migrate_from_LegacyColor_to_bevy_colorColor.md:120]: is it worth also mentioning the basic palette? I'd also specifically call out that basic::GREEN is not what Color::GREEN was? (this caught me out migrating some code)

File: release-content/0.14/migration-guides/12163_Migrate_from_LegacyColor_to_bevy_colorColor.md [hunk 3]
```diff
@@ -1,22 +1,142 @@
...
+
+#### Switch to `LinearRgba`
+
+`WireframeMaterial`, `ExtractedUiNode`, `ExtractedDirectionalLight`, `ExtractedPointLight`, `ExtractedSpotLight`, and `ExtractedSprite` now store a `LinearRgba` rather than a polymorphic `Color`. Furthermore, `Color` no longer implements `AsBindGroup`. You should store a `LinearRgba` instead to avoid conversion costs.
+
+#### When to use `Color`, `LinearRgba`, or a different color space
+
+There are a few recommendations for how to choose the right color representation:
+
+- If you are writing a game or some other executable, pick a color space and use it consistently throughout your project. Since you are the only one using your code, you can choose whichever space works best.
```
Comment (gagnus) [inline release-content/0.14/migration-guides/12163_Migrate_from_LegacyColor_to_bevy_colorColor.md:140]: I'm not sure this is good advice. Constructing colors in code (or in a data file) as HSV for example is perfectly compatible with then using them as linear rgba for lights, and sRGBA is still useful for 2D UI even when the rest of a 3D game is linear rgba.

Comment (BD103) [reply release-content/0.14/migration-guides/12163_Migrate_from_LegacyColor_to_bevy_colorColor.md:140]: Yeah, I'm considering removing this section entirely. I don't think it fits in the migration guide.

Comment (alice-i-cecile) [reply release-content/0.14/migration-guides/12163_Migrate_from_LegacyColor_to_bevy_colorColor.md:140]: Yeah, IMO this should be cut.

File: release-content/0.14/migration-guides/12163_Migrate_from_LegacyColor_to_bevy_colorColor.md (hunk 2 above)
Comment (BD103) [reply release-content/0.14/migration-guides/12163_Migrate_from_LegacyColor_to_bevy_colorColor.md:120]: Probably! I'll at least add a mention that, with the new palettes, the old colors may not be copied 1:1.

Comment (BD103) [reply release-content/0.14/migration-guides/12163_Migrate_from_LegacyColor_to_bevy_colorColor.md]: I believe I addressed this in ab89f31bbf8072b3a445aa949b42ae6fe53325bf. If you have a different suggestion, feel free to re-open this thread.

File: release-content/0.14/migration-guides/12163_Migrate_from_LegacyColor_to_bevy_colorColor.md (hunk 3 above)
Comment (BD103) [reply release-content/0.14/migration-guides/12163_Migrate_from_LegacyColor_to_bevy_colorColor.md]: Removed in ab89f31bbf8072b3a445aa949b42ae6fe53325bf :)

File: release-content/0.14/migration-guides/12163_Migrate_from_LegacyColor_to_bevy_colorColor.md [hunk 4]
```diff
@@ -1,22 +1,134 @@
...
+#### Alpha
+
+Alpha, also known as transparency, used to be referred to by the letter `a`. It is now called by its full name within structs and methods.
+
+- `Color::set_a` and `Color::a` are now `Color::set_alpha` and `Color::alpha`. These are part of the new `Alpha` trait.
+- Additionally, `Color::is_fully_transparent` is now part of the `Alpha`.
+
+#### CSS Constants
+
+The various CSS color constants are no longer stored directly on `Color`. Instead, they’re defined in the `Srgba` color space, and accessed via `bevy::color::palettes`. Call `.into()` on them to convert them into a `Color` for quick debugging use. Please note that these palettes are not necessarily 1:1 with the constants defined previously as somes names and colors have been changed. If you need the same color as before, consider copying the constant into your own code.
```
Comment (alice-i-cecile) [inline release-content/0.14/migration-guides/12163_Migrate_from_LegacyColor_to_bevy_colorColor.md:120]:
```suggestion
The various CSS color constants are no longer stored directly on `Color`. Instead, they’re defined in the `Srgba` color space, and accessed via `bevy::color::palettes`. Call `.into()` on them to convert them into a `Color` for quick debugging use. Please note that these palettes are not necessarily 1:1 with the constants defined previously as some names and colors have been changed. If you need the same color as before, consider copying the constant into your own code.
```

**Review by alice-i-cecile — APPROVED** [review]

**Pull request merged**